### PR TITLE
Add recurring opening hours logic - DDFFORM 662

### PR DESCRIPTION
**Pull request by kasperbirch1**
#### Link to issue
https://reload.atlassian.net/browse/DDFFORM-662
**Info**: https://reload.atlassian.net/browse/DDFFORM-632

#### Description
This pull request adds logic for the inclusion of recurring opening hours. It updates our openAPI specification from dpl-cms and refactors the logic to handle the new field repetition.

a checkbox has been added when adding an opening hour, which enables creating a recurring opening time. An extra modal has also been included, which shows the user the series they are creating and asks if they are sure they want to approve it. 

Additionally, have i also updated our wiremock data to ensure consistency between the editor and frontend.

#### Screenshot of the result
You cannot rely on the design of the 'Are you sure' modal, as Drupal has some styling for their tables.

https://github.com/danskernesdigitalebibliotek/dpl-react/assets/49920322/86bbb07a-9768-43a2-879e-16dbe77d7cd3

### CODE REVIEW

File: orval.config.ts [hunk 1]
```diff
@@ -139,7 +139,7 @@ export default defineConfig({
     },
     input: {
       target:
-        "https://raw.githubusercontent.com/danskernesdigitalebibliotek/dpl-cms/opening-hours-api/openapi.json",
+        "https://raw.githubusercontent.com/danskernesdigitalebibliotek/dpl-cms/opening-hours-repeated/openapi.json",
```
Comment (kasperbirch1) [inline orval.config.ts:142]: This must be set to the `develop` branch after https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/1027 is merged

File: .docker/wiremock/cms/mappings/opening-hours-all-b0c2a2f9-7b4d-4a6d-8324-13051e946c02.json
Comment (kasperg) [inline .docker/wiremock/cms/mappings/opening-hours-all-b0c2a2f9-7b4d-4a6d-8324-13051e946c02.json:1]: :+1: For renaming and restructuring these files.

File: src/apps/opening-hours-editor/EventForm.tsx [hunk 2]
```diff
@@ -4,124 +4,246 @@
...
+          disabled={isSameTime}
+        >
+          {t("openingHoursEventFormSubmitText")}
+        </button>
+        {children}
+      </form>
+
+      <Dialog closeDialog={closeDialog} ref={dialogRef}>
+        {dialogContent}
+      </Dialog>
```
Comment (kasperg) [inline src/apps/opening-hours-editor/EventForm.tsx:222]: Please consider what is going on with this dialog within the form.

As far as I can tell it is not used. `dialogContent` is never assigned.

File: src/apps/opening-hours-editor/OpeningHoursEditorEventContent.tsx [hunk 3]
```diff
@@ -10,15 +11,25 @@ const OpeningHoursEditorEventContent: React.FC<
...
       <div>
-        {extractTime(event.start)} - {extractTime(event.end)}
+        <b>{event.title}</b>
+        <div>
+          {extractTime(event.start)} - {extractTime(event.end)}
+        </div>
       </div>
+      {isSeries && (
+        <div className="opening-hours-editor-event-content__series-icon">
+          <img src={logo} alt="" className="" />
```
Comment (kasperg) [inline src/apps/opening-hours-editor/OpeningHoursEditorEventContent.tsx:30]:
```suggestion
          <img src={logo} alt="" />
```

As I see it there is no reason to have an empty classname property.

Please consider adding an alt text to the image describing that this is a repeated event. In other words: This is not a decorative image.

File: src/apps/opening-hours-editor/helper.ts [hunk 4]
```diff
@@ -10,24 +13,50 @@ const formatDateTimeString = (date: string, time: string): string => {
...
+      title: event.title,
+      color: event.color
+    },
+    date: startDate.format("YYYY-MM-DD"),
+    start_time: startDate.format("HH:mm"),
+    end_time: endDate.format("HH:mm"),
+    repetition: event.repetition,
+    // set to id 0 to because the API requires a branch_id.
+    // This will be overwritten when the event is added or edited in the useOpeningHoursEditor hook
+    branch_id: 0
```
Comment (kasperg) [inline src/apps/opening-hours-editor/helper.ts:52]: :+1:

File: src/apps/opening-hours-editor/EventForm.tsx [hunk 5]
```diff
@@ -4,124 +4,246 @@
...
+            >
+              {t("openingHoursEventFormEndDateText")}
+            </label>
+            <input
+              type="date"
+              className="opening-hours-editor-form__time-input"
+              id="event-form-end-date"
+              min={getStringForDateInput(startDate)}
+              disabled={!isRepeated}
+              required={isRepeated}
```
Comment (kasperg) [inline src/apps/opening-hours-editor/EventForm.tsx:203]: :+1: for controlling state and values here. Adding the start date as a minimum value is helpful and well as remembering to set the field as required.

File: src/apps/opening-hours-editor/ConfirmAddRepeatedOpeningHour.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { useText } from "../../core/utils/text";
+import { OpeningHoursCategoriesType } from "./types";
+
+type ConfirmAddRepeatedOpeningHourType = {
+  startDateString: string;
+  weekDayName: string;
```
Comment (kasperg) [inline src/apps/opening-hours-editor/ConfirmAddRepeatedOpeningHour.tsx:7]: Please consider only passing the startdate as a prop - and preferably as a proper date.

When we have the start date as a prop we can determine the week day name from this:

1. We already have a function for this
2. It should not be expensive to do
3. We avoid bugs where the date and weekday may come out of sync

**Review by kasperg — APPROVED** [review]
:+1: I have a few suggestions for you to look into. In general I think it looks great!

**Pull request merged**